python/cuml/test/explainer/test_gpu_treeshap.py

```diff
@@ -0,0 +1,100 @@
+#
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pytest
+import treelite
+import numpy as np
+from cuml.experimental.explainer.tree_shap import TreeExplainer
+from cuml.common.import_utils import has_xgboost
+from cuml.ensemble import RandomForestRegressor as curfr
+from sklearn.datasets import fetch_california_housing, make_classification
+
+if has_xgboost():
+    import xgboost as xgb
+
+
+@pytest.mark.parametrize('objective', ['reg:linear', 'reg:squarederror',
+                                       'reg:squaredlogerror',
+                                       'reg:pseudohubererror'])
+@pytest.mark.skipif(not has_xgboost(), reason="need to install xgboost")
+def test_xgb_regressor(objective):
+    X, y = fetch_california_housing(return_X_y=True)
+    dtrain = xgb.DMatrix(X, label=y)
+    param = {'max_depth': 8, 'eta': 0.1, 'objective': objective}
+    num_round = 10
+    xgb_model = xgb.train(param, dtrain, num_boost_round=num_round,
+                          evals=[(dtrain, 'train')])
+    tl_model = treelite.Model.from_xgboost(xgb_model)
+
+    explainer = TreeExplainer(model=tl_model)
+    out = explainer.shap_values(X)
+    correct_out = xgb_model.predict(dtrain, pred_contribs=True)
+    np.testing.assert_almost_equal(out, correct_out, decimal=5)
+
+
+@pytest.mark.parametrize('objective,n_classes',
+                         [('binary:logistic', 2),
+                          ('binary:hinge', 2),
+                          ('binary:logitraw', 2),
+                          ('count:poisson', 4),
+                          ('rank:pairwise', 5),
+                          ('rank:ndcg', 5),
+                          ('rank:map', 5),
+                          ('multi:softmax', 5),
+                          ('multi:softprob', 5)],
+                         ids=['binary:logistic', 'binary:hinge',
+                              'binary:logitraw', 'count:poisson',
+                              'rank:pairwise', 'rank:ndcg', 'rank:map',
+                              'multi:softmax', 'multi:softprob'])
+@pytest.mark.skipif(not has_xgboost(), reason="need to install xgboost")
```

Comment:
Is the runtime of this test reasonable?

Reply:
I've reduced `n_samples` from 1000 to 100, so the test should run 10x faster.